Automattic/wp-calypso
# Checkout: Add "Is this purchase for business" checkbox for certain US states

## Proposed Changes

This PR adds a new checkbox to the billing info we collect in checkout for tax purposes for certain US states (OH and CT). It reads "Is this purchase for business?" and will be used to help us collect taxes in those states.

<img width="611" alt="Screenshot 2024-03-23 at 2 01 51 PM" src="https://github.com/Automattic/wp-calypso/assets/2036909/5841568b-7e12-4c88-ab36-650c6f03bffa">

> [!NOTE]
> The tax calculation doesn't yet do anything with this information. That will come later.

Depends on D142843-code

Part of https://github.com/Automattic/payments-shilling/issues/1914

## Testing Instructions

- Sandbox the API and apply D142843-code.
- Visit checkout with a product in your cart.
- In the billing info step, enter "United States" as the country and "23000" as the postal code.
- When you reach the submit button, verify that you do not see the new checkbox.
- Edit the billing info step, enter "United States" as the country and "43000" as the postal code.
- Verify that the new checkbox appears above the submit button.
- Check and uncheck the box and verify that the cart updates (you'll see a brief loading period).
- Repeat the above step with your browser's network tab open; verify that the `/me/shopping-cart` requests made by toggling the box include a `tax.location.is_for_business` property that matches what the checkbox shows.
